src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Contains the main logic for responses by Duke.
+ */
+public class Parser {
+    /**
+     * Executes operations on the task list based on given input and
+     * returns the response by the program.
+     *
+     * @param input Input from user.
+     * @param tasks List of tasks.
+     * @return Program response as a String.
+     * @throws DukeException If input is not recognised.
+     */
+    public static String process(String input, TaskList tasks) throws DukeException {
+        if (input.equals("bye")) {
+            return null;
+        }
+
+        StringBuilder output = new StringBuilder();
+        if (input.equals("list")) {
+            output.append("  Tasks in list:\n");
+            for (int i = 0; i < tasks.size(); i++) {
+                output.append("  ").append(i + 1).append(".").append(tasks.get(i)).append("\n");
+            }
+        } else if (input.split(" ")[0].equals("done")) {
+            int inputNumber;
+            try {
+                inputNumber = Integer.parseInt(input.split(" ")[1]);
+            } catch (Exception ex) {
+                throw new DukeException("  Please provide an integer.");
+            }
+
+            try {
+                tasks.markAsDone(inputNumber - 1);
+            } catch (IndexOutOfBoundsException ex) {
+                throw new DukeException("  That number is invalid.");
+            } catch (IOException ex) {
+                throw new DukeException("  Unable to save tasks.");
+            }
+
+            output.append("  Successfully marked as done:\n");
```

Comment:
Perhaps you could try exploring the use of "\t" to create tabs, instead of coding the spaces out.